### PR TITLE
Feat/add sites addresses

### PR DESCRIPTION
**Pull request by pazepaze**

### CODE REVIEW

File: src/main/resources/db/migration/V1_0_0_4__create_sites.sql [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    add name varchar(255);
+
+alter table addresses
+    alter column partner_id drop not null;
+
+alter table addresses
+    add constraint fk_addresses_on_sites foreign key (site_id) references sites;
+
+alter table addresses
+    add bpn varchar(255) not null;
```
Comment (pazepaze) [inline src/main/resources/db/migration/V1_0_0_4__create_sites.sql:35]: note that this would require that we clear the current db

Comment (pazepaze) [inline src/main/resources/db/migration/V1_0_0_4__create_sites.sql:35]: note that this would require that we clear the current db

**Review by nicoprow — COMMENTED** [review]
Looks good, just open topics regarding business requirements

File: src/main/kotlin/com/catenax/gpdm/dto/request/BusinessPartnerSearchRequest.kt [hunk 2]
```diff
@@ -2,5 +2,6 @@ package com.catenax.gpdm.dto.request
 
 data class BusinessPartnerSearchRequest (
     val partnerProperties: BusinessPartnerPropertiesSearchRequest,
-    val addressProperties: AddressPropertiesSearchRequest
+    val addressProperties: AddressPropertiesSearchRequest,
```
[the remark this answers is not in the document]
Comment (pazepaze) [reply src/main/kotlin/com/catenax/gpdm/dto/request/BusinessPartnerSearchRequest.kt:5]: done

File: src/main/resources/db/migration/V1_0_0_4__create_sites.sql (hunk 1 above)
Comment (nicoprow) [reply src/main/resources/db/migration/V1_0_0_4__create_sites.sql:35]: that's alright. Data can be restored from CDQ

**Review by nicoprow — COMMENTED** [review]
Everything fine but still waiting on the Address Name decision: nullable or default value (what kind of default value)

File: src/main/kotlin/com/catenax/gpdm/dto/request/AddressRequest.kt [hunk 3]
```diff
@@ -8,6 +8,10 @@ import io.swagger.v3.oas.annotations.media.Schema
 
 @Schema(name = "Address Request", description = "New localized address record for a business partner")
 data class AddressRequest (
+    @Schema(description = "Business Partner Number")
+    val bpn: String?,
+    @Schema(description = "Address name")
+    val name: String?,
```
[the remark this answers is not in the document]
Comment (pazepaze) [reply src/main/kotlin/com/catenax/gpdm/dto/request/AddressRequest.kt]: address name was removed

File: src/main/kotlin/com/catenax/gpdm/dto/request/SiteRequest.kt [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+import io.swagger.v3.oas.annotations.media.ArraySchema
+import io.swagger.v3.oas.annotations.media.Schema
+
+@Schema(name = "Site Request", description = "New site record")
+data class SiteRequest(
+    @Schema(description = "Business Partner Number")
+    val bpn: String?,
+    @Schema(description = "Site name")
+    val name: String?,
```
[the remark this answers is not in the document]
Comment (pazepaze) [reply src/main/kotlin/com/catenax/gpdm/dto/request/SiteRequest.kt]: site name not nullable now

**Pull request merged**